### ISSUE TITLE
Add SAML support

### ISSUE THREAD

**Issue by bartjkdp** [Pleio/pleio_account#3]
We would like to add SAML support in order to enable login through external sources like corporate login.

**Comment by markwooff** [comment]
We have been looking at SAML & OpenID Connect for an authentication solution that could be used amongst all the services that our instance of `pleio_account` would connect to. Ideally the solution would allow for any new internal & even external services to be added without difficulty. This is why OpenID Connect seemed a right path to follow:

![image](https://user-images.githubusercontent.com/5591488/35158270-98afdc90-fd04-11e7-988b-1e69878da038.png)

**Comment by bartjkdp** [comment]
@markwooff OpenID Connect support is great! I think we will switch to OpenID connect for our Pleio services as well. So you have one Pleio account service linking to for example Elgg, Moodle, Mediawiki etc. through OpenID connect.

Additionally we offer Single-Sign-On for users from certain organizations. This means they are not required to register and activate a Pleio account, but they can use their existing organization account (mostly an Active Directory account) to create a Pleio account and instantly log in to it. So the total flow could look like:

Login request from Elgg site (OpenID Connect) -> Redirect to pleio_account -> Single Sign On request at organization (SAML2) -> Redirect back to pleio_account -> redirect back to Elgg site.

**Comment by markwooff** [comment]
Makes sense :) we may have to do something similar with connecting to an LDAP user store. The ideal solution as you mentioned is to remove the duplication in creating new user accounts for the user.
